### DISCUSSION TITLE
Default SpaceROS DDS

### DISCUSSION THREAD

**Discussion by xfiderek** [space-ros/space-ros#149] [Q&A]
Does anybody know why the default DDS used in SpaceROS is Cyclone? Default DDS in ROS2 is FastDDS, so I believe this change was deliberate, and I wonder what was the reason.

**Comment by ivanperez-keera** [comment]
Idk but I recently spoke with someone who needed to use Cyclone for a real project. So IMO we should support it. I don't know if anything would need to be done to keep supporting Cyclone if we switch to FastDDS.

**Reply by Bckempa** [reply]
@ivanperez-keera ROS2 makes switching out DDS implementations real easy, we can do this as a test config

**Comment by Bckempa** [comment]
I have no idea why the ACO picked Cyclone, that was already in place when I joined. I agree we should test with whichever DDS is blessed as the default for the underlying ROS2.

**Comment by 130s** [comment]
Related to this topic, I'm stuck with `colcon build mars_rover` (on `osrf/space-ros:humble-2024.07.0` Docker image) where I'm seeing colcon complaining about `CycloneDDS`. Posted at [robotics.stackexchange.com#112945](https://robotics.stackexchange.com/questions/112945/rmw-implementation-has-been-specified-as-rmw-fastrtps-cpp-via-but-rmw-i).  Would appreciate if anyone can take a look. I'm afraid I guess there are a lot of pitfalls there.

**Comment by asimonov** [comment]
I have tried to keep track of DDS discussions on main ROS2 forums over last few years. from my understanding FastRTPS is default because eProsima is quite active in the community and have been supporting ROS2 from very beginning.
But last couple of years there were some discussions about DDS performance, especially for big messages like point clouds and images.
Some people published some benchmarks and they were basically inconclusive. eProsima was saying that FastRTPS was not configured properly etc. (e.g. https://zenoh.io/blog/2023-03-21-zenoh-vs-mqtt-kafka-dds/)

From what I remember couple of projects that actually deal with real world chose Cyclone.
Here is nav2 take on why they use Cyclone on Humble: https://discourse.ros.org/t/nav2-issues-with-humble-binaries-due-to-fast-dds-rmw-regression/26128

I have a task for myself to revisit those publication and see if some other DDS is better.

But I would say Cyclone is probably better our of the box, from my subjective memory.

Would be interesting to hear why Cyclone is space-ros default DDS...

**Comment by asimonov** [comment]
another writeup from real project why they chose Cyclone: https://bit-bots.de/en/2022/07/experiences-with-ros-2-on-our-robots/

to me it looks like there were issues with FastRTPS around Humble release/afterwards that caused a few projects to switch to Cyclone

**Comment by patrickelectric** [comment]
Maybe Zenoh ?
